### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2019' in position 14560183: ordinal not in range(128)

### ISSUE THREAD

**Issue by ened** [nexB/scancode-toolkit#132]
While analyzing the complete linux kernel copyright situation, the following error occurred while generating the template file:

```
$ scancode -c -l -f ../zlink/docs/scancode-toolkit/template.html . ~/Desktop/linux-copyright-holders.html
Scanning files...
  [####################################]  41256                                                       
Scanning done.
Traceback (most recent call last):
  File "/Users/ened/git/ened.scancode-toolkit/bin/scancode", line 9, in <module>
    load_entry_point('scancode-toolkit==1.3.8', 'console_scripts', 'scancode')()
  File "/Users/ened/git/ened.scancode-toolkit/lib/python2.7/site-packages/click/core.py", line 664, in __call__
    return self.main(*args, **kwargs)
  File "/Users/ened/git/ened.scancode-toolkit/src/scancode/utils.py", line 61, in main
    standalone_mode=standalone_mode, **extra)
  File "/Users/ened/git/ened.scancode-toolkit/lib/python2.7/site-packages/click/core.py", line 644, in main
    rv = self.invoke(ctx)
  File "/Users/ened/git/ened.scancode-toolkit/lib/python2.7/site-packages/click/core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/ened/git/ened.scancode-toolkit/lib/python2.7/site-packages/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "/Users/ened/git/ened.scancode-toolkit/src/scancode/cli.py", line 245, in scancode
    save_results(results, format, input, output_file)
  File "/Users/ened/git/ened.scancode-toolkit/src/scancode/cli.py", line 345, in save_results
    output_file.write(as_template(results, template=format))
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2019' in position 14560183: ordinal not in range(128)
```

Using latest GIT version (+plus patches for #123 and #127).

**Comment by pombredanne** [comment]
Can you figure out which file triggered this? 
Is there any special encoding in your custom template? Did this run on a machine with a non-UTF-8 locale?

**Comment by pombredanne** [comment]
Some background on this:
http://jinja.pocoo.org/docs/dev/api/#unicode
It could be that some strings in the license object are not tracked as unicode.

**Comment by ened** [comment]
It is s bit cumbersome to figure that out because of missing stack traces, but I will try to cut it down somehow. It happened in the Android Linux Kernel folder, in arch/arm/, detail analysis to follow.

**Comment by pombredanne** [comment]
True, that is cumbersome and that should not be, sorry for that. Note that if you run `scancode --verbose` then you will get all the paths listed on screen instead of a progress bar which is likely a better tracing option.

**Comment by ened** [comment]
Not really, because scancode the first pass inside scancode will finish completely and then send the complete tree over to the "renderer" function `as_template`. I have no way to detect this beside manual finding the faulty sources.

**Comment by pombredanne** [comment]
ok, let me see if I could get something to help.

**Comment by pombredanne** [comment]
@ened I have never been able to reproduce and you did not came up with additional examples, so I am closing this. Feel free to reopen of course if this shows up again.

**Comment by pombredanne** [comment]
I am reopening this because of new developments in #390

**Comment by pombredanne** [comment]
This is now fixed with #407
